pkg/controller/deployment/deployment_controller.go

```diff
@@ -543,7 +543,18 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return dc.syncStatusOnly(d)
 	}
 
-	err = dc.classifyReplicaSets(deployment)
+	// Run cleanup policy before adopting if there is no rollback request that needs to happen. Doing so
+	// decouples the cleanup policy from the strategies where it gets executed and may block because the
+	// strategy can't proceed.
```

Comment:
Would you rephrase a bit to make it more clear what is adopted, what gets executed, and what may block?